### PR TITLE
Storyteller: Save screenshots separately

### PR DESCRIPTION
**Pull request by val-fed**

### CODE REVIEW

File: tools/storyteller/StoryWriter.java [hunk 1]
```diff
@@ -134,5 +147,9 @@ private String getCurrentTimeString() {
   private String getAbsUnsharedStoriesFolderPath() {
     return Storyteller.getUnsharedStoriesAbsPath(config);
   }
+
+  private String getId() {
```
[the remark this answers is not in the document]
Comment (val-fed) [reply tools/storyteller/StoryWriter.java]: Done.

File: tools/storyteller/StoryWriter.java [hunk 2]
```diff
@@ -134,5 +147,9 @@ private String getCurrentTimeString() {
...
+    return getCurrentTimestamp() + "_" + UUID.randomUUID().toString().split("-")[0];
```
[the remark this answers is not in the document]
Comment (val-fed) [reply tools/storyteller/StoryWriter.java]: Added a comment about it.

File: tools/storyteller/Storyteller.java [hunk 3]
```diff
@@ -33,8 +34,11 @@
   // Number of most recent shared stories to output
   private static final int RECENT_SHARED_STORIES_COUNT = 10;
 
-  private static final String FIRESTORE_STORYTELLER_ROOT = "/storyteller";
+  private static final String FIRESTORE_STORYTELLER_ROOT = "/storyteller-new";
+  private static final String FIRESTORE_SCREENSHOT_COLLECTION = "screenshots";
+  private static final String FIRESTORE_STORIES_COLLECTION = "stories";
```
[the remark this answers is not in the document]
Comment (val-fed) [reply tools/storyteller/Storyteller.java]: Done.

File: tools/storyteller/StoryReader.java [hunk 4]
```diff
@@ -30,71 +31,55 @@ public StoryReader(FileUtils fileUtils) {
...
     File file = fileUtils.getFile(absFilename);
     long fileSize = file.length();
     if (fileSize < MAX_SCREENSHOT_SIZE_BYTES) {
       try {
-        result = ByteString.copyFrom(Files.readAllBytes(file.toPath()));
+        screenBuilder
+            .setFilename(absFilename.substring(absFilename.lastIndexOf('/') + 1))
+            .setScreenshot(ByteString.copyFrom(Files.readAllBytes(file.toPath())));
       } catch (IOException e) {
         e.printStackTrace();
```
[the remark this answers is not in the document]
Comment (val-fed) [reply tools/storyteller/StoryReader.java]: Done.

File: tools/storyteller/StoryReader.java [hunk 5]
```diff
@@ -30,71 +31,55 @@ public StoryReader(FileUtils fileUtils) {
...
-    ByteString result = ByteString.EMPTY;
+  private Screenshot getScreenshot(String absFilename) {
+    Screenshot.Builder screenBuilder = Screenshot.newBuilder();
     File file = fileUtils.getFile(absFilename);
     long fileSize = file.length();
     if (fileSize < MAX_SCREENSHOT_SIZE_BYTES) {
       try {
-        result = ByteString.copyFrom(Files.readAllBytes(file.toPath()));
+        screenBuilder
+            .setFilename(absFilename.substring(absFilename.lastIndexOf('/') + 1))
```
[the remark this answers is not in the document]
Comment (val-fed) [reply tools/storyteller/StoryReader.java]: Done.

File: tools/storyteller/StoryReader.java [hunk 6]
```diff
@@ -30,71 +31,55 @@ public StoryReader(FileUtils fileUtils) {
...
-      } else {
-        absPaths = fileUtils
-            .listContents(absPath)
-            .stream()
-            .filter(file -> file.endsWith(".prototxt"))
-            .sorted()
-            .map(filename -> fileUtils.joinPaths(absPath, filename))
-            .collect(Collectors.toList());
+    List<String> absFilenames = new ArrayList<>();
+    if (state.equals(StoriesState.UNSHARED)) {
```
[the remark this answers is not in the document]
Comment (val-fed) [reply tools/storyteller/StoryReader.java]: Thank you! Done.

File: tools/storyteller/StoryWriter.java [hunk 7]
```diff
@@ -31,22 +33,32 @@
...
+      FileUtils fileUtils,
+      StoryReader reader,
+      AuthService authService) {
     this.config = storytellerConfig.getConfig();
     this.fileUtils = fileUtils;
     this.allStories =
         new ArrayList<>(
             reader.getUnsharedStories(
-                fileUtils.joinPaths(getAbsUnsharedStoriesFolderPath()), false));
+                fileUtils.joinPaths(getAbsUnsharedStoriesFolderPath())));
```
[the remark this answers is not in the document]
Comment (val-fed) [reply tools/storyteller/StoryWriter.java]: It doesn't make sense here. It stayed from the previous code. Removed it. Thanks!

File: tools/storyteller/Storyteller.java [hunk 8]
```diff
@@ -110,12 +115,18 @@ private String getUnsharedStoriesAbsPath() {
    * This method shares stories to Firebase and moves them to the shared folder.
    */
   public void share() {
-    // TODO: Add the ability to save stories larger than 1 MB to Firebase.
-    //  Upload the images themselves to Firebase Storage and
-    //  just store the filename in `StoryList`.
+    final String firestoreAuthor = FIRESTORE_STORYTELLER_ROOT + "/" + author + "/";
```
[the remark this answers is not in the document]
Comment (val-fed) [reply tools/storyteller/Storyteller.java]: Done.

File: tools/storyteller/Storyteller.java [hunk 9]
```diff
@@ -110,12 +115,18 @@ private String getUnsharedStoriesAbsPath() {
...
     StoryList storyList = StoryList.newBuilder().addAllStory(reader.getUnsharedStories(
-        getUnsharedStoriesAbsPath(), true)).build();
-    firestoreClient.addProtoDocumentToCollection(FIRESTORE_STORYTELLER_ROOT, storyList);
+        getUnsharedStoriesAbsPath())).build();
+    firestoreClient.addProtoDocumentToCollection(
+        firestoreAuthor + FIRESTORE_STORIES_COLLECTION,
```
[the remark this answers is not in the document]
Comment (val-fed) [reply tools/storyteller/Storyteller.java]: It was unreadable. It's `storyteller/data/user/<user_email>/story` now.

File: tools/storyteller/data.proto [hunk 10]
```diff
@@ -57,19 +57,26 @@ message FileData {
...
+  bytes screenshot = 2;
+}
+
 message StoryItem {
   int64 timeMs = 1;
-  bytes screenshot = 2;
-  string oneliner = 3;
-  string note = 4;
-  string screenshot_filename = 5;
+  string oneliner = 2;
```
[the remark this answers is not in the document]
Comment (val-fed) [reply tools/storyteller/data.proto]: Done.

File: tools/storyteller/data.proto [hunk 11]
```diff
@@ -57,19 +57,27 @@ message FileData {
...
 
+message Screenshot {
+  string filename = 1;
+  bytes screenshot = 2;
+}
+
 message StoryItem {
   int64 timeMs = 1;
-  bytes screenshot = 2;
+  bytes screenshot = 2 [deprecated = true];
```
Comment (oferb) [inline tools/storyteller/data.proto]: Sorry, I meant `reserved`:
https://developers.google.com/protocol-buffers/docs/proto#reserved
`reserved 2;`

File: tools/storyteller/StoryWriter.java [hunk 12]
```diff
@@ -33,6 +33,7 @@
 
   private final Config config;
   private final FileUtils fileUtils;
+  // User's email
```
Comment (oferb) [inline tools/storyteller/StoryWriter.java]: Great commenting this.

File: tools/storyteller/StoryWriter.java [hunk 13]
```diff
@@ -148,7 +148,11 @@ private String getAbsUnsharedStoriesFolderPath() {
     return Storyteller.getUnsharedStoriesAbsPath(config);
   }
 
-  private String getId() {
+  /* Returns the first element of UUID.
+   * UUID represents a 128-bit long value that is unique.
+   * It is represented in 32 Hexadecimal characters and is mixed with 4 dashes characters.
```
Comment (oferb) [inline tools/storyteller/StoryWriter.java]: `s/and is mixed with 4 dashes characters/separated by 4 dashes`

Comment (val-fed) [reply tools/storyteller/StoryWriter.java]: Done.

File: tools/storyteller/data.proto (hunk 11 above)
Comment (val-fed) [reply tools/storyteller/data.proto]: Done.

**Comment by val-fed** [comment]
closed #88

**Pull request merged**